test/test_kubeclient.rb

```diff
@@ -38,6 +38,28 @@ def test_no_version_passed
     assert_equal 'http://localhost:8080/api/v1', rest_client.url.to_s
   end
 
+  def test_pass_proxy
+    uri = URI::HTTP.build(port: 8080)
+    uri.hostname = 'localhost'
+    proxy_uri = URI::HTTP.build(host: 'myproxyhost', port: 8888)
+    client = Kubeclient::Client.new(uri, http_proxy_uri: proxy_uri.to_s)
+    rest_client = client.rest_client
+    assert_equal proxy_uri.to_s, rest_client.options[:proxy]
+
+    watch_client = client.watch_pods
+    assert_equal(watch_client.send(:build_client_options)[:proxy][:proxy_address],
+                 proxy_uri.hostname)
+    assert_equal watch_client.send(:build_client_options)[:proxy][:proxy_port], proxy_uri.port
+  end
+
+  def test_proxy_enabled
```

Comment:
It's not working.  I uploaded it to show that it fails. It seems like webmock / vcr can't catch the use of the CONNECT method with the proxy.